GrpcReader: Drain the stream when closing it.

mathieu-plak opened the pull request:
* Our Reader loop is mishandling the stream, and doesn't go up to the EOF when it has read enough data, meaning the server side connection is never closed and we leak resources.

* It's still a good practice to drain it in close, meaning no misuse of the API can lead to a leak (think rd.Read([10]byte)).

ericfaurot commented on storage/grpc_storage.go:
```diff
@@ -188,7 +189,14 @@ func (g *grpcChunkReader) Read(p []byte) (int, error) {
 // full content of the Reader. We do not actually forward the Close call over
 // GRPC, but we should at some point.
```
I think should tweak the comment here. I don't see a practical reason why we would want to forward the close over grpc.
To my understanding, if we ever wanted to interrupt the stream it would have to be done asynchronously using bidirectional streams, which wouldn't make things easier.

mathieu-plak replied:
Totally agreed

ericfaurot reviewed (approved):
with a comment

mathieu-plak committed:
GrpcReader: Drain the stream when closing it.

* Our Reader loop is mishandling the stream, and doesn't go up to the
  EOF when it has read enough data, meaning the server side connection
  is never closed and we leak resources.

* It's still a good practice to drain it in close, meaning no misuse of
  the API can lead to a leak (think rd.Read([10]byte)).
